SalesforceCommerceCloud/pwa-kit
# @W-16797872 Add Domain attribute to dw_dnt cookie

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1) Go to https://aaa-jeremy-wasatch-jeremy.mobify-storefront-staging.com/
- (step2) Open developer tools -> Application -> Cookies
- (step3) See that the cookie was created with the Domain, `.aaa-jeremy-wasatch-jeremy.mobify-storefront-staging.com`. The leading dot indicate that the cookie will be accessible to subdomains of this hostname.
- (step4) Compare this to the dw_dnt cookie being created in https://runtime-admin-n16.mobify-storefront.com/, which does not have the leading dot. Click out of the form to have the cookie be created.


The usage of the hook in the UI after this change, to enable subdomain in PWA kit for the DNT cookie is:
```
// use-dnt-notification.js
const [hostname, setHostname] = useState(undefined)
const {dntStatus, updateDNT} = useDNT(hostname)

useEffect( () => {
        if (window.location.hostname) {
            setHostname(window.location.hostname)
        }
}, [])

const onCloseNotification = () => {
        updateDNT(null, hostname)
        onClose()
}
const onClickAccept = () => {
        updateDNT(true, hostname)
        onClose()
}
const onClickDeny = () => {
        updateDNT(false, hostname)
        onClose()
}
```
# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
